BismarkUnchained/scenes/Player/Player.gd

```diff
@@ -60,14 +65,54 @@ func move_translate(delta):
 
 func move_rotate():
 	if controller:
-		var direction = Vector2(Input.get_joy_axis(device, 2), Input.get_joy_axis(device, 3))
+		var direction = Vector2(Input.get_joy_axis(device, 3), Input.get_joy_axis(device, 4))
 		if direction.length() > 0.5:
+			last_dir = Vector2(direction.x, -direction.y)
+			print(last_dir)
 			rotate_dir(direction)
 	else:
 		var direction = get_global_mouse_position() - get_global_position()
+		last_dir = Vector2(direction.x, -direction.y)
 		rotate_dir(direction)
 		
 
+func shoot():
+	if Input.is_action_pressed("player_shoot"):
+		if get_node("/root/Main").time_elapsed > time_last_shot + time_to_shoot:
+			time_last_shot = get_node("/root/Main").time_elapsed
+			var sprite = bullet_sprite.instance()
+			var x_vel = 0
+			var y_vel = 0
+			#No div by 0
+			if last_dir.x == 0:
+				if last_dir.y == 0:
+					x_vel = 1
+					y_vel = 0
+				else:
+					x_vel = 0
+					y_vel = last_dir.y
+			#Trig
+			else:
+				print(atan(last_dir.y / last_dir.x))
+				if last_dir.x < 0 && last_dir.y < 0:
```

Comment:
I'm fairly certain what you're looking for is GDScript's `atan2` ([here](https://docs.godotengine.org/en/3.1/classes/class_@gdscript.html#class-gdscript-method-atan2)) - it takes into account sign